speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/SQLDatabaseStorageTests.cpp

```diff
@@ -4,46 +4,51 @@
-
-/// @brief             Tests a database creation fail
-/// @param  p_fileName The filename
-void FailToCreateDatabase(const std::string& p_fileName)
+/// @brief connects to a database and inserts test data. Then closes database
+/// @param p_password password to connect to database
+/// @param p_inputFile test data to insert
+void OpenStoreCloseDatabase(const std::string& p_password, const char* p_inputFile)
 {
     SQLDatabaseStorage sqlDatabaseStorage;
+    sqlDatabaseStorage.OpenDatabase("127.0.0.1", 3306, "root", p_password, "test");
 
-    GET_CREATED_DATABASE_PATH
-    std::ifstream databaseFile(databasePath);
-    ASSERT_FALSE(databaseFile.good());
+    std::string path(ROOT_FOLDER "\\test_data\\testSimulationData");
+    if (!FindFileDirectory(path, p_inputFile)) throw std::exception("Can't find test files");
+
+    sqlDatabaseStorage.StoreData(path + "\\" + p_inputFile);
+    sqlDatabaseStorage.CloseDatabase();
 }
 
-/// @brief             Tests a correct creation of a database
-/// @param  p_fileName The filename
-void CreateCorrectDatabase(const std::string& p_fileName)
+/// @brief tests if the connecting, storing, and closing happens without any issues
+/// @param p_password password of database to connect to
+/// @param p_inputFile test data to store
+void DatabaseTest(const std::string& p_password, const char* p_inputFile)
 {
-    SQLDatabaseStorage sqlDatabaseStorage;
+    ASSERT_NO_THROW(OpenStoreCloseDatabase(p_password, p_inputFile));
+}
 
-    ASSERT_NO_THROW(sqlDatabaseStorage.StoreData(p_fileName));
+/// @brief tests if all database functionality happens fast enough
+/// @param p_password password of database to connect to
+/// @param p_inputFile test data to store
+void DatabaseTimeTest(const std::string& p_password, const char* p_inputFile)
+{
+    ASSERT_DURATION_LE(2, DatabaseTest(p_password, p_inputFile))
+}
 
-    GET_CREATED_DATABASE_PATH
-    std::ifstream databaseFile(databasePath);
-    ASSERT_TRUE(databaseFile.good());
-    databaseFile.close();
-    remove(databasePath.c_str());
+/// @brief tests if an exception is thrown when the input data is incorrect
+/// @param p_password password of database to connect to
+/// @param p_inputFile test data to store
+void CatchDatabaseError(const std::string& p_password, const char* p_inputFile)
+{
+    ASSERT_THROW(OpenStoreCloseDatabase(p_password, p_inputFile), std::exception);
 }
 
-TEST_CASE(SQLDatabaseStorageTests, CorrectInputTest, CreateCorrectDatabase, ("correctInput.txt"))
+#define YOUR_PASSWORD [YOUR PASSWORD HERE]
 
-TEST_CASE(SQLDatabaseStorageTests, NoStartTest, FailToCreateDatabase, ("noStart.txt"))
-TEST_CASE(SQLDatabaseStorageTests, NonExistingKeyTest, FailToCreateDatabase, ("nonExistingKey.txt"))
-TEST_CASE(SQLDatabaseStorageTests, NoVariableNamesTest, FailToCreateDatabase, ("noVariableNames.txt"))
+//TEST_CASE(SQLDatabaseStorageTests, InitialiseDatabase, DatabaseTest, (YOUR_PASSWORD, "test_file.txt"));
```

Comment:
Are these still needed or can they be removed?

Reply:
These can't be run on github because mysql is required, but they can run locally, so they can stay